src/test/kotlin/uk/gov/dluhc/emsintegrationapi/service/PostalVoteApplicationServiceTest.kt

```diff
@@ -0,0 +1,122 @@
+import uk.gov.dluhc.emsintegrationapi.config.ApiProperties
+import uk.gov.dluhc.emsintegrationapi.database.entity.RecordStatus
+import uk.gov.dluhc.emsintegrationapi.database.repository.PostalVoteApplicationRepository
+import uk.gov.dluhc.emsintegrationapi.mapper.PostalVoteMapper
+import uk.gov.dluhc.emsintegrationapi.models.PostalVote
+import uk.gov.dluhc.emsintegrationapi.testsupport.testdata.buildPostalVoteApplication
+import java.util.stream.IntStream
+
+@ExtendWith(MockitoExtension::class)
+internal class PostalVoteApplicationServiceTest {
+
+    @Mock
+    private lateinit var apiProperties: ApiProperties
+
+    @Mock
+    private lateinit var postalVoteMapper: PostalVoteMapper
+
+    @Mock
+    private lateinit var postalVoteApplicationRepository: PostalVoteApplicationRepository
+
+    @InjectMocks
+    private lateinit var postalVoteApplicationService: PostalVoteApplicationService
+
+    private val defaultPageSize = 100
+
+    @Nested
+    inner class PageSizeIsNotProvided {
+        @AfterEach
+        fun afterEach() {
+            verify(apiProperties).defaultPageSize
+        }
+        @Test
+        fun `should return maximum of 100 postal vote applications`() {
+            // Given
+            validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 100, pageSizeRequested = null)
+        }
+
+        @Test
+        fun `system does not have requested number of records in the DB`() {
+            // Given validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 100, pageSizeRequested = null)
+            validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 10, pageSizeRequested = null)
+        }
+
+        @Test
+        fun `system does not have any records`() {
+            validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 0, pageSizeRequested = null)
+        }
+    }
+
+    @Nested
+    inner class PageSizeIsProvided {
+        @AfterEach
+        fun afterEach() {
+            verifyNoInteractions(apiProperties)
+        }
+
+        @Test
+        fun `should return maximum of 100 postal vote applications`() {
+            // Given
+            validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 100, pageSizeRequested = 200)
+        }
+
+        @Test
+        fun `system does not have requested number of records in the DB`() {
+            // Given validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 100, pageSizeRequested = null)
+            validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 10, pageSizeRequested = 100)
+        }
+
+        @Test
+        fun `system does not have any records`() {
+            validateFetchPostalVoteApplications(numberOfRecordsToBeReturned = 0, pageSizeRequested = 100)
+        }
+    }
+
+    private fun validateFetchPostalVoteApplications(numberOfRecordsToBeReturned: Int, pageSizeRequested: Int?) {
+        // Given
+
+        if (pageSizeRequested == null) {
```

Comment:
need to be careful about using if statements in tests, ideally best to avoid

Reply:
Happy to change it, just a reusable function for conditional stubbing

Reply:
Done